# This patch fixes the in-place operation bug in ECC.
# It was developed by the upstream, but not yet available in
# the latest release.
--- a/ecc-add-eh.c	2017-10-01 15:30:55.210597670 -0700
+++ b/ecc-add-eh.c	2017-10-01 15:30:17.123251715 -0700
@@ -98,8 +98,8 @@
   ecc_modp_mul (ecc, x3, B, z1);
 
   /* y3 */
-  ecc_modp_mul (ecc, B, F, C); /* ! */
-  ecc_modp_mul (ecc, y3, B, z1);
+  ecc_modp_mul (ecc, B, F, z1); /* ! */
+  ecc_modp_mul (ecc, y3, B, C); /* Clobbers z1 in case r == p. */
 
   /* z3 */
   ecc_modp_mul (ecc, B, F, G);
